resources/frameworks.yaml

```diff
@@ -32,9 +32,26 @@ AutoGluon_bestquality:
   extends: AutoGluon
   description: |
     AutoGluon with 'best_quality' preset provides the most accurate overall predictor.
+    Refer to https://auto.gluon.ai/stable/tutorials/tabular_prediction/tabular-quickstart.html#presets for a description of the presets.
   params:
     presets: best_quality
 
+AutoGluon_hq:
+  extends: AutoGluon
+  description: |
+    AutoGluon with 'high_quality' preset provides a very strong predictor with 8x+ faster inference speed than 'best_quality'.
+    Refer to https://auto.gluon.ai/stable/tutorials/tabular_prediction/tabular-quickstart.html#presets for a description of the presets.
+  params:
+    presets: high_quality
+
+AutoGluon_gq:
+  extends: AutoGluon
+  description: |
+    AutoGluon with 'good_quality' preset provides a strong predictor with 16x+ faster inference speed than 'best_quality'.
+    Refer to https://auto.gluon.ai/stable/tutorials/tabular_prediction/tabular-quickstart.html#presets for a description of the presets.
+  params:
+    presets: good_quality
```

Comment:
@sebhrusen it feels like this is getting unwieldy. Should we extend the configuration format to more easily allow for different presets?

Reply:
well, I don't have the feeling that the configuration format is the problem here (or is it? do you have sth in mind like a specific CLI syntax only for presets? sth that would be simple enough to work for various frameworks?)

I mean users can create as many framework definitions in their user `frameworks.yaml`, we don't have to accept all of those in the default `frameworks.yaml`.

Reply:
currently, CLI syntax allows passing framework params like
`-Xf.presets=good_quality` but this works only for local runs, this is not forwarded to ec2 instances or to docker images, we need custom `frameworks.yaml` for that, but maybe this could be improved (whitelisting?) although I'm not sure I like the idea of having params coming from different sources when executed remotely.

Reply:
I don't really know what I had in mind at the time. But if every framework does this, it will blow up the configuration file many times over. MLJar has 4 presets, GAMA has more presets, and I suspect (and hope) more frameworks might follow.

A custom parameter which is specifically for the mode/preset might be reasonable, considering it's a "special" hyperparameter in the sense that it's the only one we allow for benchmarking. Consider the last three additional entries in the framework definition:

```yaml
AutoGluon:
  version: "stable"
  description: |
    AutoGluon-Tabular: Unlike existing AutoML frameworks that primarily focus on model/hyperparameter selection,
    AutoGluon-Tabular succeeds by ensembling multiple models and stacking them in multiple layers.
  project: https://auto.gluon.ai
  refs: [https://arxiv.org/abs/2003.06505]
  presets: ['best_quality', 'high_quality', 'good_quality']
  preset: best_quality    # could also be the first of `modes`  by convention
  param_name: presets  # the framework name for the hyperparameter, so that we can easily convert internally to add it to `params` to avoid changing integration scripts
```
This could be complimented by a CLI argument which overrides the `mode` argument. It wouldn't solve the issue of forwarding.

On the other hand, it requires additional logic for something that works well already (even if it's minimal).

Reply:
maybe the first mistake was to allow non-default presets in the `resources/frameworks.yaml`.
The tool offers the possibility for anyone to add custom configurations, we're not going to limit that, and for the paper's benchmarks, we use the `franeworks_yyyyQn.yaml` files where we can decide to use non-default presets after agreeing with each team ("was it a good idea to allow multiple configs for those?" is a separate question).
The problem with making `preset` special is that we're officially encouraging this approach (which I like btw), and, as we already see, it creates issues on our side (multiplication, way to fine tune the framework only for the  benchmark, in contradiction with the original approach of benchmarking all tools with their default setup...).

That's why personally, I'd go backwards and accept only default setup in `frameworks.yaml`, rather than creating a new convoluted syntax. And we would decide "customizations" only for the `franeworks_yyyyQn.yaml` ones.
My second choice would be to just add a param/section to the framework description whitelisting framework params that are always forwarded, it's relatively simple to do and will just allow the usual CLI syntax also for docker and aws.